### PR TITLE
[gherkin-c] Linking against libmath

### PR DESCRIPTION
**Pull request by sappo**
## Problem

Llinking fails on log10 function in error_list.c

## Solution

Include libmath in the target_link_libraries so log10 function can be linked.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

### CODE REVIEW

File: gherkin/c/CMakeLists.txt [hunk 1]
```diff
@@ -70,11 +71,11 @@ target_include_directories(gherkin PUBLIC
     "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include;${CMAKE_CURRENT_SOURCE_DIR}/src>")
 
 add_executable(gherkinexe ${GHERKIN_CLI})
-target_link_libraries(gherkinexe gherkin)
+target_link_libraries(gherkinexe gherkin m)
```
Comment (mpkorstanje) [inline gherkin/c/CMakeLists.txt:74]:
```suggestion
target_link_libraries(gherkinexe gherkin PRIVATE m)
```
I don't know too much about C or CMake. But shouldn't this be linked privately so as not to expose math as part of the link interface?

File: gherkin/c/CMakeLists.txt [hunk 2]
```diff
@@ -70,11 +71,11 @@ target_include_directories(gherkin PUBLIC
...
 target_include_directories(gherkinexe PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include ${CMAKE_CURRENT_SOURCE_DIR}/src)
 
 add_executable(gherkin_generate_tokens ${GENERATE_TOKEN})
-target_link_libraries(gherkin_generate_tokens gherkin)
+target_link_libraries(gherkin_generate_tokens gherkin m)
```
Comment (mpkorstanje) [inline gherkin/c/CMakeLists.txt:78]:
```suggestion
target_link_libraries(gherkin_generate_tokens gherkin PRIVATE m)
```

**Comment by mpkorstanje** [comment]
Thanks for the suggestion. We are currently a bit short handed to really look into this but at a glance it looks al right.

File: gherkin/c/CMakeLists.txt (hunk 1 above)
Comment (sappo) [reply gherkin/c/CMakeLists.txt]: @mpkorstanje correct but we need to link the library privately against math not the executables. I just pushed another commit that fixes this issue. Thanks!

**Comment by sappo** [comment]
@mpkorstanje correct but we need to link the library privately against math not the executables. I just pushed another commit that fixes this issue. Thanks!

**Comment by aslakhellesoy** [comment]
Hi @sappo,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Comment by mpkorstanje** [comment]
Cheers!

**Pull request merged**